CMakeLists.txt

```diff
@@ -125,10 +125,11 @@ if(WIN32)
     dlfcn-win32
     GIT_REPOSITORY https://github.com/dlfcn-win32/dlfcn-win32.git
     GIT_TAG v1.4.1
+    SOURCE_SUBDIR use-an-unexist-path-to-skip-cmake-of-dlfcn-win32
     EXCLUDE_FROM_ALL)
   FetchContent_MakeAvailable(dlfcn-win32)
   target_include_directories(mlx PRIVATE "${dlfcn-win32_SOURCE_DIR}/src")
-  target_link_libraries(mlx PRIVATE dl)
+  target_sources(mlx PRIVATE ${dlfcn-win32_SOURCE_DIR}/src/dlfcn.c)
```

Comment:
it might be simpler to `set(BUILD_SHARED_LIBS OFF)` ?

Reply:
Updated the code to set `BUILD_SHARED_LIBS`, also bumped the required cmake version for using `block`.